management/daemon.py

```diff
@@ -39,14 +40,23 @@ def newview(*args, **kwargs):
 		error = None
 		try:
 			email, privs = auth_service.authenticate(request, env)
-		except ValueError as e:
-			# Authentication failed.
+		except totp.MissingTokenError as e:
 			privs = []
-			error = "Incorrect username or password"
+			error = str(e)
+		except totp.BadTokenError as e:
+			# Write a line in the log recording the failed login
+			log_failed_login(request)
 
+			privs = []
```

Comment:
What about moving ``privs = []`` before the try statement? Then you could remove it in the except clauses :) If the login is valid, the privs list will be filled by auth_service.authenticate, otherwise it will stay empty and you do not need to reset the array in the except block

Reply:
Good catch, thanks.